### ISSUE TITLE
TTC + YVP Stay - wrong total number of nights

### ISSUE THREAD

**Issue by iswara108** [danthareja/sivananda-price-calculator#1]
When calculating a TTC stay followed by a YVP stay, the total number of nights is not taking the TTC length into account.

For example - a person takes TTC in November, and stays for another 10 nights, the price code should be taken as 21+ nights for the YVP stay, but it's taking the 7-13 price code instead.

**Comment by danthareja** [comment]
Hmm @iswara108, I can't seem to repro this one... Can you provide a specific failing case?

Here's the test case I have passing already:

```js
const calculator = new SivanandaPriceCalculator({
  adults: 1,
  stays: [{
    type: 'TTC',
    roomId: 'TENT_HUT',
    checkInDate: '2017-04-03',
    checkOutDate: '2017-05-03'
  }, {
    type: 'ROOM',
    roomId: 'TENT_HUT',
    checkInDate: '2017-05-03',
    checkOutDate: '2017-05-11'
  }]
})
expect(calculator.getTotalNumberOfNights()).to.equal(38)
expect(calculator.getGrandTotal()).to.equal(4362-7*8)
```

Additionally, the UI seems to be working (for beachfront at least)
![screen shot 2017-11-03 at 9 47 25 am](https://user-images.githubusercontent.com/6980359/32382917-2a4b76fc-c07c-11e7-93c2-ca412668e715.png)

**Comment by iswara108** [comment]
Yes, please see this case:
http://local.sivanandabahamas.org/price_calculator/#adults=1&children=0&stays%5B0%5D%5Btype%5D=TTC&stays%5B0%5D%5BttcId%5D=NOVEMBER_2017&stays%5B0%5D%5BroomId%5D=TENT_HUT&stays%5B0%5D%5BcheckInDate%5D=2017-11-03&stays%5B0%5D%5BcheckOutDate%5D=2017-11-03&stays%5B0%5D%5BroomDiscount%5D%5Btype%5D=PERCENT&stays%5B0%5D%5BroomDiscount%5D%5Bvalue%5D=0&stays%5B0%5D%5ByvpDiscount%5D%5Btype%5D=PERCENT&stays%5B0%5D%5ByvpDiscount%5D%5Bvalue%5D=0&stays%5B1%5D%5Btype%5D=ROOM&stays%5B1%5D%5BroomId%5D=TENT_HUT&stays%5B1%5D%5BcheckInDate%5D=2017-12-03&stays%5B1%5D%5BcheckOutDate%5D=2017-12-04&stays%5B1%5D%5BroomDiscount%5D%5Bvalue%5D=10&stays%5B1%5D%5BroomDiscount%5D%5Btype%5D=PERCENT&stays%5B1%5D%5ByvpDiscount%5D%5Bvalue%5D=10&stays%5B1%5D%5ByvpDiscount%5D%5Btype%5D=PERCENT

It should select the price code of 21+ nights for the last night.

Om
Iswara Chaitanya

**Comment by danthareja** [comment]
@iswara108, fixed in https://github.com/danthareja/sivananda-price-calculator-ui/commit/8f5a3f2be9eb94cc42587ccbdcd2bde8a3d1125a

Just a heads up, you'll have to regenerate the scenario to see the fix.

Clicking the link above will still take you to a broken version, because the broken dates are hardcoded into the url

**Comment by iswara108** [comment]
Om Brahmaswaroop

Are you sure this one is closed?
Perhaps it is not published yet to the
local.sivanandabahamas.org/price_calculator/ ?

**Comment by danthareja** [comment]
yes, i believe so @iswara108. Recreating the previous error gave me this link which seems to be working

http://local.sivanandabahamas.org/price_calculator/#adults=1&children=0&stays%5B0%5D%5Btype%5D=ROOM&stays%5B0%5D%5BroomId%5D=TENT_HUT&stays%5B0%5D%5BcheckInDate%5D=2017-12-03&stays%5B0%5D%5BcheckOutDate%5D=2017-12-04&stays%5B0%5D%5BroomDiscount%5D%5Bvalue%5D=10&stays%5B0%5D%5BroomDiscount%5D%5Btype%5D=PERCENT&stays%5B0%5D%5ByvpDiscount%5D%5Bvalue%5D=10&stays%5B0%5D%5ByvpDiscount%5D%5Btype%5D=PERCENT&stays%5B1%5D%5Btype%5D=TTC&stays%5B1%5D%5BttcId%5D=NOVEMBER_2017&stays%5B1%5D%5BroomId%5D=TENT_SPACE&stays%5B1%5D%5BcheckInDate%5D=2017-11-03&stays%5B1%5D%5BcheckOutDate%5D=2017-12-03&stays%5B1%5D%5BroomDiscount%5D%5Btype%5D=PERCENT&stays%5B1%5D%5BroomDiscount%5D%5Bvalue%5D=0&stays%5B1%5D%5ByvpDiscount%5D%5Btype%5D=PERCENT&stays%5B1%5D%5ByvpDiscount%5D%5Bvalue%5D=0

Can you confirm? If it is not working, can you give me a repro case?

**Comment by iswara108** [comment]
I'm not sure I get it.

When I open this calculation:
http://local.sivanandabahamas.org/price_calculator/#adults=1&children=0&stays%5B0%5D%5Btype%5D=TTC&stays%5B0%5D%5BttcId%5D=NOVEMBER_2017&stays%5B0%5D%5BroomId%5D=TENT_HUT&stays%5B0%5D%5BcheckInDate%5D=2017-11-03&stays%5B0%5D%5BcheckOutDate%5D=2017-11-03&stays%5B0%5D%5BroomDiscount%5D%5Btype%5D=PERCENT&stays%5B0%5D%5BroomDiscount%5D%5Bvalue%5D=0&stays%5B0%5D%5ByvpDiscount%5D%5Btype%5D=PERCENT&stays%5B0%5D%5ByvpDiscount%5D%5Bvalue%5D=0&stays%5B1%5D%5Btype%5D=ROOM&stays%5B1%5D%5BroomId%5D=TENT_HUT&stays%5B1%5D%5BcheckInDate%5D=2017-12-03&stays%5B1%5D%5BcheckOutDate%5D=2017-12-04&stays%5B1%5D%5BroomDiscount%5D%5Bvalue%5D=10&stays%5B1%5D%5BroomDiscount%5D%5Btype%5D=PERCENT&stays%5B1%5D%5ByvpDiscount%5D%5Bvalue%5D=10&stays%5B1%5D%5ByvpDiscount%5D%5Btype%5D=PERCENT

I expect the total number of nights to be 31, not 1.
Therefore the price of the one extra night should be lower (taking the 21+
nights price code)

Om
Iswara Chaitanya

**Comment by danthareja** [comment]
This is expected due to the way the url loading works. If you look closely at the link you provided:

...type%5D=TTC&stays%5B0%5D%5BttcId%5D=NOVEMBER_2017&stays%5B0%5D%5BroomId%5D=TENT_HUT&stays%5B0%5D%5B**checkInDate%5D=2017-11-03**&stays%5B0%5D%5B**checkOutDate%5D=2017-11-03**...

The dates are hardcoded into the link. This one shows a check in date and check out date of the same day, so the UI will just display what the link says.

The link I posted was slightly different:

...type%5D=TTC&stays%5B1%5D%5BttcId%5D=NOVEMBER_2017&stays%5B1%5D%5BroomId%5D=TENT_SPACE&stays%5B1%5D%5B**checkInDate%5D=2017-11-03**&stays%5B1%5D%5B**checkOutDate%5D=2017-12-03**..

Showing the correct dates for the TTC and therefore the correct dates in the UI.

If you click on the same link you posted, drop the TTC stay, and the re-add it, you should see the correct number of days. Please do not trust any old links with TTC stays, and instead re-generate them using the method I just described.

Does this clear things up @iswara108?

**Comment by iswara108** [comment]
Ahhh....

I get it. Amazing.
Thank you very much.
